redisson/src/main/java/org/redisson/config/Config.java

```diff
@@ -361,7 +361,7 @@ protected MasterSlaveServersConfig getMasterSlaveServersConfig() {
         return masterSlaveServersConfig;
     }
 
-    protected void setMasterSlaveServersConfig(MasterSlaveServersConfig masterSlaveConnectionConfig) {
+    public void setMasterSlaveServersConfig(MasterSlaveServersConfig masterSlaveConnectionConfig) {
```

Comment:
All these methods should remain protected to avoid confusion for developers and force them to use "use...." methods instead.

Reply:
For Spring Boot to read the configuration into an object, it uses setters by default. There is another way to do this, by reading in the several configs in the RedissonProperties class, but then I still need a way to somehow set the server configuration in the `Config` object. Would it be a possibility to make the 
```
SingleServerConfig useSingleServer(SingleServerConfig config)
```
public instead of default visibility?

Otherwise, I only see a possibility in making a custom mapper that will set each field in the server configs separately, but this will have some impact on the maintainability of the project, since adding a field to a server config requires you to add a mapping for that as well.